### PR TITLE
github-ci: enable ubuntu-20.04 hosts for packaging

### PR DESCRIPTION
**Pull request by avtikhon**
Patch set consists of 2 commits:

1. Created local composite action 'pack_and_deploy' for Tarantool packages
    creation and deployment. It was created using local scripts in packages
    workflows. It let to change common parts of packages creations and
    deployment in each packaging workflow to call for this action. It helped
    to consolidate all the instructions on packages creation and deployment
    in a single place.

2. Enabling ubuntu-20.04 hosts for packaging workflows found that DEB
    package Github Actions workflows do not need to install createrepo
    tool. Also found that createrepo is not ready for ubuntu-20.04 as
    described in (till ubuntu-21.04 where it is available as the new
    version of this tool named as 'createrepo_c' as DEB package):

  3a7c21023367424baa9bf2d4d823102cb2b3c751 ('github-ci: ubuntu-20.04 misses createrepo package')

To fix it added 'createrepo_c' build and installation from sources
and changed in update_repo tool 'createrepo' tool to 'createrepo_c'.
This patch is needed to use these workflows on self-hosted runners
which run under ubuntu-20.04 by default for now.

Also checking the patch on ubuntu-20.04 hosts got the following issue:
```
  Regenerated DEB file: pool/xenial/main/t/tarantool/tarantool-common_2.8.0.185.g4c3e0eb-1_all.deb

  <botocore.awsrequest.AWSRequest object at 0x7f7998a4ca90>

  <botocore.awsrequest.AWSRequest object at 0x7f627d965070>
  make: *** [.gitlab.mk:131: deploy] Error 255
  Error: Process completed with exit code 2.
```
Found that there is already issue exists in Github Actions on it [1].
Provided solution to setup AWS region in environment helped to workaround the issue [2].

Closes tarantool/tarantool-qa#110
Closes tarantool/tarantool-qa#111

[1]: https://github.com/aws/aws-cli/issues/5234
[2]: https://github.com/aws/aws-cli/issues/5234#issuecomment-635459464

### CODE REVIEW

**Comment by Totktonada** [comment]
But how we'll run centos / fedora jobs on Ubuntu 20.04?

**Comment by avtikhon** [comment]
> But how we'll run centos / fedora jobs on Ubuntu 20.04?

I've changed the patch to use ubuntu-20.04 for all package workflows.

File: .gitlab.mk [hunk 1]
```diff
@@ -128,11 +128,14 @@ package: deploy_prepare
 
 deploy:
 	echo ${GPG_SECRET_KEY} | base64 -d | gpg --batch --import || true
-	./tools/update_repo.sh -o=${OS} -d=${DIST} \
-		-b="${LIVE_REPO_S3_DIR}/${BUCKET}" build
-	if [ "${CI_COMMIT_TAG}" != "" ]; then \
+	# found that libcreaterepo_c.so installed in local lib path
+	export LD_LIBRARY_PATH=/usr/local/lib ; \
```
Comment (Totktonada) [inline .gitlab.mk:132]: How about something like [this](https://stackoverflow.com/a/49524393/1598057)? Or, better, setup the variable in the action, where we build the tool?

File: .github/actions/pack_and_deploy/action.yml [hunk 2]
```diff
@@ -17,7 +17,24 @@ runs:
             startsWith(github.ref, 'refs/heads/2.') ||
             startsWith(github.ref, 'refs/tags') ) }} ; then
           sudo apt-get -y update
-          sudo apt-get install -y procmail createrepo awscli reprepro
+          sudo apt-get install -y procmail awscli reprepro
+          # For RPM packages need to build 'createrepo_c' tool from
+          # sources till Github Actions will use ubuntu-21.04 where
+          # it be avaliable as DEB package for installation.
+          if [ ${OS} == 'el' -o ${OS} == 'fedora' -o ${OS} == 'opensuse-leap' ] ; then
+            git clone https://github.com/rpm-software-management/createrepo_c.git
```
Comment (Totktonada) [inline .github/actions/pack_and_deploy/action.yml:25]: I would checkout a tag, `master` may change.

File: tools/update_repo.sh [hunk 3]
```diff
@@ -657,7 +657,7 @@ function pack_rpm {
     done
 
     # create the new repository metadata files
-    createrepo --no-database --update --workers=2 \
+    createrepo_c --no-database --update --workers=2 \
```
Comment (Totktonada) [inline tools/update_repo.sh:660]: Do you looked over the [list of changes](https://github.com/rpm-software-management/createrepo_c#differences-in-behavior-between-createrepo_c-and-createrepo) to spot possible negative consequences?

**Review by Totktonada — APPROVED** [review]
Looked briefly. No hard objections from me.

File: tools/update_repo.sh (hunk 3 above)
Comment (avtikhon) [reply tools/update_repo.sh:660]: Right, I've checked it, but main confidence that it works correct, was after I've checked that all packages successfully saved and were available to install from temporary S3 YUM repositories. Check the following branch test workflows:
https://github.com/tarantool/tarantool/tree/avtikhon/gh-qa-110-ci-deb-tmp
and S3 repository:
s3://tarantool_repo/check_live

File: .github/actions/pack_and_deploy/action.yml (hunk 2 above)
Comment (avtikhon) [reply .github/actions/pack_and_deploy/action.yml:25]: Right, thought the same, but forgot to set, changed master to 0.17.1 (latest).

File: .gitlab.mk (hunk 1 above)
Comment (avtikhon) [reply .gitlab.mk:132]: I've tried to set it to action, but it failed work for update_repo tool. But using your example I've successfully checked it using export as target:
https://github.com/tarantool/tarantool/runs/2288050206#step:5:13393

Comment (Totktonada) [reply .gitlab.mk:132]: > I've tried to set it in action, but it failed to work for update_repo tool.

Via $GITHUB_ENV?

**Review by LeonidVas — APPROVED** [review]
LGTM

**Pull request merged**